numba/unicode.py

```diff
@@ -531,6 +557,96 @@ def find_impl(a, b):
         return find_impl
 
 
+@overload_method(types.UnicodeType, 'rfind')
+def unicode_rfind(src, sub, start=None, end=None):
+
+    if not (start is None or isinstance(start, (types.Omitted, types.Integer, types.Optional))):
+        raise TypingError("Start arg must be of type Integer or None")
+
+    if not (end is None or isinstance(end, (types.Omitted, types.Integer, types.Optional))):
+        raise TypingError("End arg must be of type Integer or None")
+
+    if isinstance(sub, types.UnicodeType):
+        def rfind_impl(src, sub, start=start, end=end):
```

Comment:
Given the work on `str.count` to create an algorithm which is easier to maintain, https://github.com/numba/numba/pull/4204#pullrequestreview-264737562, I'm wondering if the algorithm for `rfind` is just a variation on that? i.e. instead of accumulating a count, return the index. What do you think?

Reply:
Yes that's right! It's just a variation so I have already started working on using simpler algorithm used in count method.  I'll resubmit it soon.